payjoin-relay/src/main.rs

```diff
@@ -163,56 +196,4 @@ fn not_found() -> Response<Body> {
     res
 }
 
-#[derive(Debug)]
-pub(crate) struct Buffer {
-    buffer: Arc<Mutex<Bytes>>,
-    sender: broadcast::Sender<()>,
-}
-
-impl Clone for Buffer {
-    fn clone(&self) -> Self {
-        Buffer { buffer: Arc::clone(&self.buffer), sender: self.sender.clone() }
-    }
-}
-
-impl Buffer {
-    pub fn new() -> Self {
-        let (sender, _) = broadcast::channel(1);
-        Buffer { buffer: Arc::new(Mutex::new(Bytes::new())), sender }
-    }
-
-    pub async fn push(&self, request: Bytes) {
-        debug!("push request: {} bytes", request.len());
-        let mut buffer = self.buffer.lock().await;
-        *buffer = request;
-        debug!("pushed");
-        // Send a broadcast to notify all listeners that a push has occurred
-        let _ = self.sender.send(()); // If there are no active receivers, the error is ignored
-        debug!("sent signal");
-    }
-
-    pub async fn peek(&self) -> Bytes {
-        debug!("peek");
-
-        {
-            let buffer = self.buffer.lock().await;
-            if !buffer.is_empty() {
-                return buffer.clone();
-            }
-        }
-
-        let mut receiver = self.sender.subscribe();
-        debug!("empty, awaiting push");
-
-        let _ = receiver.recv().await;
-        debug!("pushed, awaiting lock");
-
-        let buffer = self.buffer.lock().await;
-        debug!("peeked");
-        buffer.clone()
-    }
-
-    pub async fn peek_with_timeout(&self, timeout: Duration) -> Option<Bytes> {
-        tokio::time::timeout(timeout, self.peek()).await.ok()
-    }
-}
+fn shorten_string(input: &str) -> String { input.chars().take(8).collect() }
```

Comment:
hashing may provide more uniqueness than just taking the first 8, not super critical for rollout